### DISCUSSION TITLE
Finder support

### DISCUSSION THREAD

**Discussion by marakas592** [moonglow/FlashForge_Marlin#158] [General]
Hello, I accidentally found your project on the Internet. I would like to clarify a few questions:
1. As I understand it, no one has checked the compatibility of the Marlin firmware for Flashforge Finer and whether it is possible to install the firmware on it
2. What are the options for killing the printer during/due to firmware so that it is not possible to restore it on your own?

My machine basic info:

Machine Type : Flashforge Finder
Machine Name : Finder
Firmware Version : 2.8.4.366 F2.12 20210823
Board Version : 2.X
Linux Kernel: 1.0.366
Control Module : 1.0.1
Dependency Library : 1.0.1
Firmware : 2.8.4

If you need any additional info, i will provide. Thank you

**Comment by tracedgod** [comment]
Hey there @marakas592, as the firmware currently stands, I do not think the Finder is compatible with this marlin setup, and with your particular 2.x board, I'm almost certain that it wouldn't work. If my memory serves me correct, there are 2 different versions of the Finder main board, the older versions running a firmware that does not utilize the Linux kernel (sort of like the factory firmware on the supported printers in this repo), and the later revision boards that are running the Linux kernel firmware  (like your 2.x board).

I actually owned a Finder with the 2.x board before I got a Dreamer. I was able to root my Finder using a hack that went by CrashForge (looks like the repo unfortunately does not exist anymore, though I **_might_** still have a copy of it somewhere and will definitely look for it if you are interested).

Additionally, due to the fact the Finder is basically an entirely separate platform from the printers in this repo (which are all loosely based off of the Dreamer), adding support for it might be out of the question for this repo sorry to say. Your best bet would be to find a way to dump the bootloader, and if you have some knowledge in reverse-engineering, figure out how it works and locate some entry-points for loading some custom firmware and go from there. From what I remember, I think the firmware update process definitely uses a similar process to the dreamer at the very least, and I think in the past I was able to load the factory firmware using a USB drive in the port, but it's been a while since I've owned the Finder now so I can't really say for sure.

**Reply by marakas592** [reply]
Hi, thanks for the detailed answer. I think that let the printer print with the factory firmware. Experiments will not lead to good things

**Reply by marakas592** [reply]
I also forgot to ask, if you had finder, were you somehow able to solve the issue with fan control? I mean the flashforge fan fix by Moonglow.

What about crashforge, I don’t quite understand what possibilities this will give?

**Reply by tracedgod** [reply]
not 100% sure on the fan fix, not really positive I had an issue with the fan on mine. 

CrashForge basically just allows you to get root access to the printer essentially. since it's Linux based, this just means you can mess around with the internal files on the printer, as well as some provided plugins that came with CrashForge like a Samba Server, an MJPEG stream (for a custom camera install if you had one), etc.

**Reply by marakas592** [reply]
Can you share crashforge files?

**Reply by qwertychouskie** [reply]
If the board is running Linux, I feel that Klipper might be a better choice than Marlin.

**Comment by Lushors-maker** [comment]
Hello everyone, does anyone know where the old firmware for the Flashforge Finder 3D Printer (model 1) can be downloaded?
